examples/mixed_signal/tests/test_regulator_plot.py

```diff
@@ -0,0 +1,74 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import cocotb
+from cocotb.triggers import Timer
+import math
+import matplotlib.pyplot as plt
+
+bit_toggle = 0
```

Comment:
Don't use globals. You had a testbench class and had a bit toggle generator on the object. I'd do that same thing for this test.

Reply:
I explicitly also wanted testcases without classes to show it's also possible (mixed-signal engineers are usually rather far from more advanced software concepts), but I solved this now differently - just read the old value of the toggle and invert it.

Reply:
> just read the old value of the toggle and invert it

Even simpler, excellent.